src/main/java/org/folio/service/finance/transaction/EncumbranceService.java

```diff
@@ -302,20 +322,22 @@ private String buildEncumbrancesByOrderQuery(String orderId) {
       + AND + "encumbrance.sourcePurchaseOrderId==" + orderId;
   }
 
-  private String buildUnreleasedEncumbrancesByOrderQuery(String orderId) {
+  private String buildUnreleasedEncumbrancesByOrderQuery(String orderId, String fiscalYearId) {
     return ENCUMBRANCE_CRITERIA
-                + AND + "encumbrance.sourcePurchaseOrderId==" + orderId
-                    + AND + "encumbrance.status <> " + Encumbrance.Status.RELEASED;
+      + AND + "encumbrance.sourcePurchaseOrderId==" + orderId
+      + AND + "encumbrance.status <> " + Encumbrance.Status.RELEASED
+      + AND + "fiscalYearId == " + fiscalYearId;
   }
 
   private String buildEncumbrancesByPoLineQuery(String polineId) {
     return ENCUMBRANCE_CRITERIA + AND + "encumbrance.sourcePoLineId==" + polineId;
   }
 
-  private String buildUnreleasedEncumbrancesByPoLineQuery(String polineId) {
+  private String buildUnreleasedEncumbrancesByPoLineQuery(String polineId, String fiscalYearId) {
     return ENCUMBRANCE_CRITERIA
```

Comment:
ENCUMBRANCE_CRITERIA is the first here and in this case we are doing almost full scan on the table, if transactions has for example 1 mil recorods and 300k are encumbrances - so pg is doing full scan on 300k that is not performant.
Instead we have indexes already for encumbrance.sourcePurchaseOrderId and encumbrance.sourcePoLineId.
https://github.com/folio-org/mod-finance-storage/blob/master/src/main/resources/templates/db_scripts/schema.json#L419
So we should move this criteria to beginning of the condition and in this case appropriate index should be used and opening order time should be increased a lot

Reply:
[MODFISTO-466](https://folio-org.atlassian.net/browse/MODFISTO-466) would help with that, if we add an index for `transactionType` (which I think we should given how frequently we use it).

Reply:
even if we add that index for transactionType I am not sure that index for encumbrance.sourcePurchaseOrderId and encumbrance.sourcePoLineId will be used. Because transactionType can split our initial dataset for 7 chunks because we have 7 possible allowed values for transaction type. So if we have 700k records - we need to do full scan on 100k records after applying index for transactionType .
Are you okay if I change order to paste ENCUMBRANCE_CRITERIA to the end?

Reply:
done